stage0/run.go

```diff
@@ -165,20 +159,78 @@ func Prepare(cfg PrepareConfig, dir string, uuid *types.UUID) error {
 
 	cdoc, err := json.Marshal(cm)
 	if err != nil {
-		return fmt.Errorf("error marshalling pod manifest: %v", err)
+		return nil, fmt.Errorf("error marshalling pod manifest: %v", err)
 	}
+	return cdoc, nil
+}
 
-	log.Printf("Writing pod manifest")
-	fn := common.PodManifestPath(dir)
-	if err := ioutil.WriteFile(fn, cdoc, 0700); err != nil {
-		return fmt.Errorf("error writing pod manifest: %v", err)
+// validatePodManifest reads the use specified pod manifest, prepare the app images
+// and do validation for the pod manifest. If the pod manifest looks good, returns
```

Comment:
s/do validation for the/validates the/
if the pod manifest passes validation, it returns